.github/workflows/label-analysis.yml

```diff
@@ -13,39 +13,45 @@ jobs:
     if: github.triggering_actor != 'dd-devflow[bot]'
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        with:
-          python-version: "3.11"
-          cache: "pip"
-      - name: Install dependencies
+      - name: Check team assignment
         run: |
-          python -m pip install --upgrade pip
-          pip install -r requirements.txt
-          pip install -r tasks/libs/requirements-github.txt
-      - name: Run PR check for team assignment label
-        env:
-          BRANCH_NAME: ${{ github.ref_name }}
-          PR_ID: ${{ github.event.pull_request.number }}
-        run: |
-          inv -e lint-teamassignment
+          labels="${{join(github.event.pull_request.labels.*.name, ' ')}}"
+          for label in $(echo "$labels" | tr ' ' '\n'); do
+            if [[ "$label" =~ ^qa/ ]]; then
+              echo "A label to skip QA is set -- no need for team assignment"
+              exit 0
+            fi
+            if [[ "$label" =~ ^team/ && "$label" != team/triage ]]; then
+              echo "Team label found: $label"
+              exit 0
+            fi
+          done
+          echo "PR ${{github.event.number}} requires at least one non-triage team assignment label (label starting by 'team/')"
+          exit 1
   skip-qa:
     if: github.triggering_actor != 'dd-devflow[bot]'
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-python@v5
-        with:
-          python-version: "3.11"
-          cache: "pip"
-      - name: Install dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install -r requirements.txt
-          pip install -r tasks/libs/requirements-github.txt
-      - name: Run PR check for QA labels
-        env:
-          BRANCH_NAME: ${{ github.ref_name }}
-          PR_ID: ${{ github.event.pull_request.number }}
+      - name: Check ability to skip qa
         run: |
-          inv -e lint-skip-qa
\ No newline at end of file
+          skip=0
+          done=0
+          noChange=0
+          labels="${{join(github.event.pull_request.labels.*.name, ' ')}}"
+          for label in $(echo "$labels" | tr ' ' '\n'); do
+            if [[ "$label" == "qa/skip-qa" ]]; then
+              skip=1
+            fi
+            if [[ "$label" == "qa/done" ]]; then
+              done=1
+            fi
+            if [[ "$label" == "qa/no-code-change" ]]; then
+              noChange=1
+            fi
+          done
+          if [[ ($skip -eq 0 && $done -eq 0 && $noChange -eq 0) || ($skip -eq 1 && $((done ^ noChange)) -eq 1) ]]; then
```

Comment:
I think `($skip -eq 1 && $((done ^ noChange)) -eq 1)` could be written as `$skip -eq 1 && $done -neq $noChange` which is more readable IMO

Reply:
Hagrid